scripts/version_buckets.py

```diff
@@ -0,0 +1,53 @@
+"""
+Reads all the current buckets, ensures that versioning is switched on for those
+that don't yet have it and sets the life cycle configuration to send
+non-current versions of files to glacier storage after 30 days.
+
+This script should only be run once, when the following PR lands in master in
+the control panel:
+
+https://github.com/ministryofjustice/analytics-platform-control-panel/pull/816
+
+(From that moment on, all S3 buckets will be configured in this way by the
+control panel, as a default.)
+"""
+import boto3
+
+
+s3 = boto3.client('s3')
+buckets = s3.list_buckets()
+for bucket_dict in buckets["Buckets"]:
+    bucket_name = bucket_dict["Name"]
+    if not bucket_name.startswith("alpha"):
```

Comment:
🤔 this only runs on buckets with `alpha-` as prefix but it's still quite wide.
Don't know how else to reduce this tho so I guess will have to do but see next comment.

Reply:
OK... so what's the way to reliably identify the buckets created by the CP for user data..?

Reply:
sorry just saw this comment, good question, I thought there wasn't really a way but during standup I thought of something that may work.

so:
- [datawarehouse buckets should have a `buckettype=datawarehouse` label](https://github.com/ministryofjustice/analytics-platform-control-panel/blob/ff8589aa5aa0b7fcc1d96767290f9de74dc1c869/controlpanel/api/aws.py#L209-L210) (did some digging and [this was first introduced ~2 years ago here](https://github.com/ministryofjustice/analytics-platform-control-panel/commit/95adf0d227cedbfacea557d6ed99c4e89c2455b4) AFAICT)
- webapps buckets don't have any labels at the moment (this is also information potentially) but their name starts with `alpha-app-`

So _maybe_ if you combine these two things you _could_ infer that if an S3 Bucket starts with `alpha-app-` or has the `buckettype=datawarehouse` chances are that it was created by the Control Panel.

Now, of course it's just an idea, I can't imagine S3 buckets created beyond the CP to have similar characteristics but of course it's hard to tell.

Reply:
Thanks for the heads up. I'll have a play. :+1:

Reply:
Alternatively the control panel displays these buckets, so you could use whatever method it uses to identify them. I believe they will be listed in the db. But the tag might be easiest, if it is indeed reliable.

Reply:
Oh yeah @davidread - I guess an alternative is to extract this list from the CP and use it as a whitelist of S3 buckets to migrate, I can't really tell from the top of my head whether this would be simpler than getting this information from AWS (potentially could be simpler) but it's definitely more accurate than that heuristics.

I suppose a benefit of a whitelist is that potentially you could run on few buckets first before pulling the trigger on the whole list.

Reply:
OK, latest push implements this recipe. Please check.